### PR TITLE
Fix products specs pt.1

### PR DESCRIPTION
**Pull request by pmilewski**

### CODE REVIEW

File: frontend/spec/features/products_spec.rb [hunk 1]
```diff
@@ -96,24 +98,22 @@
 
       it 'when adding a product to the cart', js: true do
         visit spree.product_path(product)
-        click_button 'Add To Cart'
-
-        within('#main-nav-bar') do
-          click_link 'Home'
+        add_to_cart(product) do
+          visit '/cart'
```
Comment (KirillKayumov) [inline frontend/spec/features/products_spec.rb]: It opens the cart by default
https://github.com/spark-solutions/spree/blob/db0f7a57c2d2fd4115dbc2d944ecae29c2d3d5c6/frontend/spec/support/add_to_cart.rb#L12

File: frontend/spec/features/products_spec.rb [hunk 2]
```diff
@@ -96,24 +98,22 @@
...
 
       it "when on the 'address' state of the cart", js: true do
         visit spree.product_path(product)
-        click_button 'Add To Cart'
-        click_button 'Checkout'
-        fill_in 'order_email', with: 'test@example.com'
-        click_button 'Continue'
-        within('tr[data-hook=item_total]') do
+        add_to_cart(product) do
+          visit '/checkout'
```
Comment (KirillKayumov) [inline frontend/spec/features/products_spec.rb]: can we click on "Checkout" button here?

File: frontend/app/controllers/spree/products_controller.rb [hunk 3]
```diff
@@ -70,7 +70,7 @@ def load_variants
     end
 
     def load_products
-      retrieved = Rails.cache.fetch params.permit! do
+      retrieved = Rails.cache.fetch params.permit!.merge(current_currency: current_currency) do
```
Comment (KirillKayumov) [inline frontend/app/controllers/spree/products_controller.rb]: Should we update it here as well?
https://github.com/spark-solutions/spree/blob/db0f7a57c2d2fd4115dbc2d944ecae29c2d3d5c6/frontend/app/views/spree/products/index.html.erb#L39

File: frontend/spec/features/products_spec.rb [hunk 4]
```diff
@@ -8,18 +8,20 @@
   end
 
   before do
-    visit spree.root_path
+    Rails.application.routes.default_url_options[:host] = 'example.com'
```
Comment (KirillKayumov) [inline frontend/spec/features/products_spec.rb]: I believe that this line changes the host for all specs. Should we set it somewhere in a `support` file?

File: frontend/spec/features/products_spec.rb [hunk 5]
```diff
@@ -234,57 +234,52 @@
   end
 
   it 'is able to hide products without price' do
-    expect(page).to have_css('#products .product-list-item').exactly(9).times
+    expect(page).to have_css('.product-component-name').exactly(9).times
     Spree::Config.show_products_without_price = false
     Spree::Config.currency = 'CAN'
-    visit spree.root_path
-    expect(page).not_to have_css('#products .product-list-item')
+    visit '/products'
```
Comment (KirillKayumov) [inline frontend/spec/features/products_spec.rb]: Can we use route helpers here for consistency?

File: frontend/spec/features/products_spec.rb [hunk 6]
```diff
@@ -234,57 +234,52 @@
...
-                             'Ruby on Rails Baseball Jersey',
+    expect(tmp.sort!).to eq(['Ruby on Rails Baseball Jersey',
                              'Ruby on Rails Jr. Spaghetti',
-                             'Ruby on Rails Ringer T-Shirt'])
+                             'Ruby on Rails Ringer T-Shirt',
+                             'Ruby on Rails Stein'])
+  end
+
+  it 'is able to display products priced between 151 and 200 dollars across multiple pages' do
+    Spree::Config.products_per_page = 3
```
Comment (KirillKayumov) [inline frontend/spec/features/products_spec.rb]: Doesn't it change the config for all specs running after this one? If yes, we must reset it to a default value after this scenario.

File: frontend/spec/features/products_spec.rb (hunk 4 above)
Comment (damianlegawiec) [reply frontend/spec/features/products_spec.rb]: @pmilewski @KirillKayumov also this host is set somewhere else earlier

File: frontend/spec/features/products_spec.rb (hunk 6 above)
Comment (pmilewski) [reply frontend/spec/features/products_spec.rb]: It changes the config for only this one spec.

**Comment by pmilewski** [comment]
I have problems with products_spec.rb
```
33 examples, 6 failures

Failed examples:

rspec ./spec/features/products_spec.rb:234 # Visiting Products is able to hide products without price
rspec ./spec/features/products_spec.rb:294 # Visiting Products is not able to put a product without a current price in the cart
rspec ./spec/features/products_spec.rb:275 # Visiting Products is able to display products priced between 151 and 200 dollars across multiple pages
rspec ./spec/features/products_spec.rb:222 # Visiting Products product with taxons displays selected taxon in breadcrumbs
rspec ./spec/features/products_spec.rb:215 # Visiting Products product with taxons displays breadcrumbs for the default taxon when none selected
rspec ./spec/features/products_spec.rb:183 # Visiting Products a product with variants, images only for the variants does not display no image available

```

File: frontend/app/controllers/spree/products_controller.rb [hunk 7]
```diff
@@ -3,13 +3,16 @@ class ProductsController < Spree::StoreController
...
+    before_action :load_taxon, only: :index
 
     helper 'spree/taxons'
 
     respond_to :html
 
     def index
+      @searcher = build_searcher(params.merge(include_images: true))
+      @products = @searcher.retrieve_products
+      @products = @products.includes(:possible_promotions) if @products.respond_to?(:includes)
```
Comment (KirillKayumov) [inline frontend/app/controllers/spree/products_controller.rb]: Isn't `@products` always an ActiveRecord relation that always responds to `#includes`?

File: frontend/spec/features/products_spec.rb [hunk 8]
```diff
@@ -96,24 +96,20 @@
 
       it 'when adding a product to the cart', js: true do
         visit spree.product_path(product)
-        click_button 'Add To Cart'
+        add_to_cart(product)
```
Comment (KirillKayumov) [inline frontend/spec/features/products_spec.rb]: `#add_to_cart` helper visits a Product's page internally, so we don't need this `visit` on the 98th line

File: frontend/spec/features/products_spec.rb [hunk 9]
```diff
@@ -96,24 +96,20 @@
...
-        end
-
-        within('.cart-info') do
+        within('.shopping-cart-total-amount') do
           expect(page).to have_content('19.99 ₽')
         end
       end
 
       it "when on the 'address' state of the cart", js: true do
         visit spree.product_path(product)
```
Comment (KirillKayumov) [inline frontend/spec/features/products_spec.rb]: Same here

File: frontend/spec/features/products_spec.rb [hunk 10]
```diff
@@ -122,9 +118,9 @@
 
   it 'is able to search for a product' do
     fill_in 'keywords', with: 'shirt'
-    click_button 'Search'
+    first('input[type=submit]').click
```
Comment (KirillKayumov) [inline frontend/spec/features/products_spec.rb]: `fill_in 'keywords', with: 'shirt\n'` should both enter the text and "press" Enter

Comment (KirillKayumov) [reply frontend/spec/features/products_spec.rb]: Maybe `"shirt\n"` should be in double quotes to make it work

File: frontend/app/controllers/spree/products_controller.rb (hunk 7 above)
Comment (damianlegawiec) [reply frontend/app/controllers/spree/products_controller.rb]: @KirillKayumov not necessary as you can swap the searcher class for something different. This is the way it was in previous spree versions and we need to keep it so all extensions/customizations work properly

File: frontend/spec/features/products_spec.rb (hunk 8 above)
Comment (damianlegawiec) [reply frontend/spec/features/products_spec.rb]: @KirillKayumov please fix this

File: frontend/spec/features/products_spec.rb (hunk 9 above)
Comment (damianlegawiec) [reply frontend/spec/features/products_spec.rb]: @KirillKayumov same as above

File: frontend/spec/features/products_spec.rb (hunk 10 above)
Comment (damianlegawiec) [reply frontend/spec/features/products_spec.rb]: @KirillKayumov please fix this

Comment (KirillKayumov) [reply frontend/spec/features/products_spec.rb]: Unfortunately, it doesn't work. Probably it works only when a JS driver is enabled. I can't enable it for this scenario either, so I'm going to leave it as is.

**Pull request merged**